src/frontend/js/components/deleteSolutionMutation.js

```diff
@@ -0,0 +1,30 @@
+import { commitMutation, graphql } from 'react-relay'
+
+const mutation = graphql`
+  mutation deleteSolutionMutation($input: deleteSolutionInput!) {
+    deleteSolution(input: $input) {
+      solution{
+        id
+      }
+    }
+  }
+`
+
+function commit (props, solutionData) {
+  const environment = props.relay.environment
+  return commitMutation(environment, {
+    mutation,
+    variables: {
+      input: {
+        solutionId: props.id
+      }
+    },
+    onCompleted: (response, errors) => {
+      props.router.push('/solutions')
```

Comment:
Can't this behaviour be handled in the respective react components that invoke this mutation?